airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream_reader.py

```diff
@@ -46,26 +69,29 @@ def get_matching_files(
         """
         ...
 
-    @staticmethod
-    def filter_files_by_globs(files: List[RemoteFile], globs: List[str]) -> Iterable[RemoteFile]:
+    @classmethod
+    def filter_files_by_globs(cls, files: List[RemoteFile], globs: List[str]) -> Iterable[RemoteFile]:
         """
         Utility method for filtering files based on globs.
         """
         seen = set()
 
         for file in files:
-            for g in globs:
-                # Use the GLOBSTAR flag to enable recursive ** matching
-                # (https://facelessuser.github.io/wcmatch/wcmatch/#globstar)
-                if globmatch(file.uri, g, flags=GLOBSTAR):
-                    if file.uri not in seen:
-                        seen.add(file.uri)
-                        yield file
+            if cls.file_matches_globs(file, globs):
+                if file.uri not in seen:
+                    seen.add(file.uri)
+                    yield file
+
+    @staticmethod
+    def file_matches_globs(file: RemoteFile, globs: List[str]) -> bool:
+        # Use the GLOBSTAR flag to enable recursive ** matching
+        # (https://facelessuser.github.io/wcmatch/wcmatch/#globstar)
+        return any(globmatch(file.uri, g, flags=GLOBSTAR) for g in globs)
 
     @staticmethod
-    def get_prefixes_from_globs(globs: List[str]) -> List[str]:
+    def get_prefixes_from_globs(globs: List[str]) -> Set[str]:
         """
         Utility method for extracting prefixes from the globs.
         """
-        prefixes = {glob.split("*")[0].rstrip("/") for glob in globs}
```

Comment:
was the rstrip just a bug? How do we know/verify this new behavior is the expected one?

Reply:
In S3, if the trailing "/" is removed, the search wont' be restricted to a directory. For instance, the prefix `a/b` would match `a/bbbb.csv` instead of "everything under directory b". So allowing the "/" just allows us to be more restrictive about which files are requested.